### PR TITLE
chore(linters): Enable `appendCombine` checker for gocritic

### PR DESCRIPTION
**Pull request by zak-pawel**
resolves https://github.com/influxdata/telegraf/issues/14402

### CODE REVIEW

**Review by Hipska — APPROVED** [review]

File: plugins/aggregators/histogram/histogram_test.go [hunk 1]
```diff
@@ -188,8 +188,10 @@ func TestHistogramWithReset(t *testing.T) {
 // TestHistogramWithAllFields tests two metrics for one period and for all fields
 func TestHistogramWithAllFields(t *testing.T) {
 	var cfg []config
-	cfg = append(cfg, config{Metric: "first_metric_name", Buckets: []float64{0.0, 15.5, 20.0, 30.0, 40.0}})
-	cfg = append(cfg, config{Metric: "second_metric_name", Buckets: []float64{0.0, 4.0, 10.0, 23.0, 30.0}})
+	cfg = append(cfg,
+		config{Metric: "first_metric_name", Buckets: []float64{0.0, 15.5, 20.0, 30.0, 40.0}},
+		config{Metric: "second_metric_name", Buckets: []float64{0.0, 4.0, 10.0, 23.0, 30.0}},
+	)
```
Comment (srebhan) [inline plugins/aggregators/histogram/histogram_test.go:194]: Why not
```golang
	cfg := []config{
		{Metric: "first_metric_name", Buckets: []float64{0.0, 15.5, 20.0, 30.0, 40.0}},
		{Metric: "second_metric_name", Buckets: []float64{0.0, 4.0, 10.0, 23.0, 30.0}},
	}
```

File: plugins/aggregators/histogram/histogram_test.go [hunk 2]
```diff
@@ -266,8 +268,10 @@ func TestHistogramWithAllFields(t *testing.T) {
 // TestHistogramWithAllFieldsNonCumulative tests two metrics for one period and for all fields
 func TestHistogramWithAllFieldsNonCumulative(t *testing.T) {
 	var cfg []config
-	cfg = append(cfg, config{Metric: "first_metric_name", Buckets: []float64{0.0, 15.5, 20.0, 30.0, 40.0}})
-	cfg = append(cfg, config{Metric: "second_metric_name", Buckets: []float64{0.0, 4.0, 10.0, 23.0, 30.0}})
+	cfg = append(cfg,
+		config{Metric: "first_metric_name", Buckets: []float64{0.0, 15.5, 20.0, 30.0, 40.0}},
+		config{Metric: "second_metric_name", Buckets: []float64{0.0, 4.0, 10.0, 23.0, 30.0}},
+	)
```
Comment (srebhan) [inline plugins/aggregators/histogram/histogram_test.go:274]: Same as above.

File: plugins/aggregators/histogram/histogram_test.go [hunk 3]
```diff
@@ -430,8 +434,10 @@ func TestHistogramMetricExpiration(t *testing.T) {
 	}()
 
 	var cfg []config
-	cfg = append(cfg, config{Metric: "first_metric_name", Fields: []string{"a"}, Buckets: []float64{0.0, 10.0, 20.0, 30.0, 40.0}})
-	cfg = append(cfg, config{Metric: "second_metric_name", Buckets: []float64{0.0, 4.0, 10.0, 23.0, 30.0}})
+	cfg = append(cfg,
+		config{Metric: "first_metric_name", Fields: []string{"a"}, Buckets: []float64{0.0, 10.0, 20.0, 30.0, 40.0}},
+		config{Metric: "second_metric_name", Buckets: []float64{0.0, 4.0, 10.0, 23.0, 30.0}},
+	)
```
Comment (srebhan) [inline plugins/aggregators/histogram/histogram_test.go:440]: See above.

File: plugins/inputs/lustre2/lustre2_test.go [hunk 4]
```diff
@@ -317,80 +317,80 @@ func TestLustre2GeneratesJobstatsMetrics(t *testing.T) {
...
-		"jobstats_setattr":         uint64(2),
-		"jobstats_getxattr":        uint64(4),
-		"jobstats_setxattr":        uint64(5),
-		"jobstats_statfs":          uint64(1207),
-		"jobstats_sync":            uint64(3),
-		"jobstats_samedir_rename":  uint64(706),
-		"jobstats_crossdir_rename": uint64(201),
-	})
+	fields = append(fields,
+		map[string]interface{}{
```
Comment (srebhan) [inline plugins/inputs/lustre2/lustre2_test.go:321]:
```suggestion
	fields := map[string]interface{}{
```

File: plugins/inputs/lustre2/lustre2_test.go [hunk 5]
```diff
@@ -317,80 +317,80 @@ func TestLustre2GeneratesJobstatsMetrics(t *testing.T) {
...
+			"jobstats_getattr":         uint64(11),
+			"jobstats_setattr":         uint64(1),
+			"jobstats_getxattr":        uint64(3),
+			"jobstats_setxattr":        uint64(4),
+			"jobstats_statfs":          uint64(1205),
+			"jobstats_sync":            uint64(2),
+			"jobstats_samedir_rename":  uint64(705),
+			"jobstats_crossdir_rename": uint64(200),
+		},
+		map[string]interface{}{
```
Comment (srebhan) [inline plugins/inputs/lustre2/lustre2_test.go:357]: Just define all fields at once?

File: plugins/inputs/systemd_units/systemd_units.go [hunk 6]
```diff
@@ -208,10 +208,12 @@ func setSystemctl(timeout config.Duration, unitType string, pattern string) (*by
...
 		params = append(params, psplit...)
 	}
-	params = append(params, "--all", "--plain")
-	// add type as configured in config
-	params = append(params, fmt.Sprintf("--type=%s", unitType))
-	params = append(params, "--no-legend")
+	params = append(params,
+		"--all", "--plain",
+		// add type as configured in config
+		fmt.Sprintf("--type=%s", unitType),
```
Comment (srebhan) [inline plugins/inputs/systemd_units/systemd_units.go:214]:
```suggestion
		"--type="+unitType,
```

File: plugins/outputs/websocket/websocket_test.go [hunk 7]
```diff
@@ -152,8 +152,10 @@ func TestWebSocket_Write_OK(t *testing.T) {
 	connect(t, w)
 
 	var metrics []telegraf.Metric
-	metrics = append(metrics, testutil.TestMetric(0.4, "test"))
-	metrics = append(metrics, testutil.TestMetric(0.5, "test"))
+	metrics = append(metrics,
```
Comment (srebhan) [inline plugins/outputs/websocket/websocket_test.go:155]:
```suggestion
	metrics := []telegraf.Metric{
```

**Review by srebhan — COMMENTED** [review]
@zak-pawel just a few comments where we can avoid `append` altogether by directly declaring the structures...

File: plugins/aggregators/histogram/histogram_test.go (hunk 1 above)
Comment (zak-pawel) [reply plugins/aggregators/histogram/histogram_test.go]: Fixed.

File: plugins/aggregators/histogram/histogram_test.go (hunk 2 above)
Comment (zak-pawel) [reply plugins/aggregators/histogram/histogram_test.go]: Fixed.

File: plugins/aggregators/histogram/histogram_test.go (hunk 3 above)
Comment (zak-pawel) [reply plugins/aggregators/histogram/histogram_test.go]: Fixed.

File: plugins/inputs/lustre2/lustre2_test.go (hunk 4 above)
Comment (zak-pawel) [reply plugins/inputs/lustre2/lustre2_test.go]: Fixed.

File: plugins/inputs/lustre2/lustre2_test.go (hunk 5 above)
Comment (zak-pawel) [reply plugins/inputs/lustre2/lustre2_test.go]: Fixed.

File: plugins/inputs/systemd_units/systemd_units.go (hunk 6 above)
Comment (zak-pawel) [reply plugins/inputs/systemd_units/systemd_units.go]: Fixed.

File: plugins/outputs/websocket/websocket_test.go (hunk 7 above)
Comment (zak-pawel) [reply plugins/outputs/websocket/websocket_test.go]: Fixed.

**Review by srebhan — APPROVED** [review]

**Review by powersj — APPROVED** [review]

**Pull request merged**